Server 3 ga preview (#5172)
circleci/circleci-docs: jekyll/_cci2/server-3-install-prerequisites.adoc

---
= CircleCI Server v3.x Prerequisites
:page-layout: classic-docs
:page-liquid:
:icons: font
:toc: macro
:toc-title:

In order to configure the CircleCI Server application, you will need to ensure the following general and infrastructure-specific requirements are met. You will need:

* An existing Kubernetes cluster (see xref:server-3-install-creating-your-first-cluster.adoc[our guide] if you need help creating one), for example:
** Creating an Amazon EKS cluster - https://docs.aws.amazon.com/eks/latest/userguide/create-cluster.html[Amazon EKS]
*** Using `eksctl` is our recommended option, as it creates a VPC and selects the proper security group for you.
** Creating clusters - https://cloud.google.com/kubernetes-engine/docs/how-to#creating-clusters[Google GKE] +
+
Note that your Kubernetes cluster must meet the following minimum overall cluster requirements relative to the number of active
CircleCI Server users: +
+
--
[.table.table-striped]
[cols=5*, options="header", stripes=even]
|===
| Number of daily active CircleCI users
| Minimum Nodes
| Total CPU
| Total RAM
| NIC speed

| < 500
| 3
| 8 cores
| 32 GB
| 1 Gbps

| 500+
| 3
| 48 cores
| 240 GB
| 10 Gbps
|===
--

> *** Your cluster must have outbound access to pull Docker containers and verify your license. If you don’t want to give open outbound access, see our https://help.replicated.com/community/t/customer-firewalls/55[list of ports] that will need access.

Reviewer: ```suggestion
*** Your cluster must have outbound access to pull Docker containers and verify your license. If you do not want to provide open outbound access, see our https://help.replicated.com/community/t/customer-firewalls/55[list of ports] that will need access.
```

* A CircleCI License file. Contact https://support.circleci.com/hc/en-us/requests/new[CircleCI support] for a license.
* The <<Required Tools>> tools installed

> ** Note that the `kubectl` tool must also be configured to have access your cluster

Reviewer: ```suggestion
** Note that the `kubectl` tool must also be configured to have access your cluster.
```
